lib/eibflatpak.py

```diff
@@ -641,54 +628,73 @@ def _set_extra_languages(self):
         repo.write_config(repo_config)
         self.installation.drop_caches()
 
-    def enumerate_remotes(self):
-        """Enumerate all configured remotes"""
-        # Set languages since subpaths get calculated when calling
-        # installation.list_remote_related_refs_sync().
+    def _set_masked(self):
+        """Set the core.xa.masked repo option from excluded flatpaks
+
+        This is used to filter out excluded flatpaks that would
+        automatically be added as related refs.
+
+        Unfortunately, this is a global option whereas the excluded
+        flatpaks are specified in the configuration per remote. It may
+        be better to use Flatpak.Remote.set_filter(), but the semantics
+        might not be the same.
+        """
+        excluded = set()
+        for remote in self.remotes.values():
+            excluded |= remote.exclude
+        if len(excluded) == 0:
+            return
+
+        repo = self.get_repo()
+        repo_config = repo.copy_config()
+        value = ';'.join(excluded)
+        logger.info('Setting repo option core.xa.masked to %s', value)
+        repo_config.set_value('core', 'xa.masked', value)
+        repo.write_config(repo_config)
+        self.installation.drop_caches()
+
+    def _remove_masked(self):
+        """Remove the core.xa.masked repo option"""
+        repo = self.get_repo()
+        repo_config = repo.copy_config()
+        logger.info('Removing repo option core.xa.masked')
+        try:
+            repo_config.remove_key('core', 'xa.masked')
+        except GLib.Error as err:
+            # Ignore errors for missing group or key
+            if err.matches(GLib.KeyFile.error_quark(),
+                           GLib.KeyFileError.GROUP_NOT_FOUND):
+                pass
+            elif err.matches(GLib.KeyFile.error_quark(),
+                             GLib.KeyFileError.KEY_NOT_FOUND):
+                pass
```

Comment:
This is the WORST thing about this GLib API.

Reply:
In C it's not too bad. In GI bindings it's terrible.